### PR TITLE
User guide

### PR DESCRIPTION
**Pull request by trevorhardy**
### Description

If merged this pull request will update the HELICS User Guide for v2.0.

### CODE REVIEW

**Comment by trevorhardy** [comment]
Feel free to give whatever you feel is an appropriate amount of attention to the review of the User Guide. I was more liberal with my suggested reviewers than I know is necessary.

**Comment by phlptp** [comment]
I will try to review this tonight,  I will probably make a commit tonight with any edits I can make and some other review comments.  I would suggest others do the same.  It would be nice if we can tag a final tomorrow.

**Comment by phlptp** [comment]
Should this be a merge commit, or squash and merge?  @kdheepak @nightlark @trevorhardy

**Comment by trevorhardy** [comment]
I think we can get away with a squash merge. The history of the development of the User Guide is not important in the same way code development is.

**Comment by kdheepak** [comment]
We can do a squash and merge. @trevorhardy unless you absolutely need it we won't keep the `user_guide` branch. You can maintain one separately if you'd like, and all the commits should be accessible from this page anyway.

File: docs/user-guide/filters.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+     Because HELICS values do not pass through this generic network, they cannot be operated on by filters. Since HELICS values are used to represent physics of the system not the control and coordination of it, it is appropriate that filters not be available to modify them. It is entirely possible to use HELICS value federates to send control signals to other federates; co-simulations can and have been made to work in such ways. Doing so, though, cuts out the possibility of using filters and, as we'll see, the easy integration of communication system simulators.
+  
+The figure below is an example of a representation of the message topology of a generic co-simulation federation composed entirely of message federates. Source and destination filters have been implemented (indicated by the blue endpoints), each showing a different built-in HELICS filter function. 
+  
+  * As a reminder, a single endpoint can be used to both send and receive messages (as shown by Federate 4) but a single filter will only act on messages sent or messages received. Both a source filter and a destination filter can be set up on a single endpoint.
```
Comment (phlptp) [inline docs/user-guide/filters.md]: With HELICS 2.0 filters can act on as many endpoints as desired, sending or receiving.  I wasn't sure how to specify that in this list.    Filters have destination targets and source targets.    There are no longer separate filter types for source and destination.

File: docs/user-guide/helics_co-sim_sequence.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+5. **Time Grant** - Based on the time requests and grants from all the connected federates, a core will determine the next time it can grant to a federate to guarantee none of the federates will be asked to simulate a point back in time. That is, if the time coordinator is doing its job correctly, every federate will receive a time that is the same as or larger than the last time it was granted. HELICS does support a configuration and some other situations that allows a federate to break this rule but this is a very special situation and would require all the federates to support this jumping back in time, or accept non-causality and some randomness in execution. This and other aspects of granting time are discussed in more detail in [the section on timing](./timing.md).
```
Comment (phlptp) [inline docs/user-guide/helics_co-sim_sequence.md]: I modified this section quite a bit.  It didn't really convey one of the key things about HELICS in that there is no central timekeeper.  Everything is distributed, each federate has there own timeCoordinator which makes the grant decision based on the connections.  This section previously implied a centralized time keeping algorithm  which was fundamentally false.    The brokers really have very little to do with the time.

**Comment by trevorhardy** [comment]
@phlptp thanks for the clarifications and improvements. This is exactly the kinds of things I was hoping you'd provide.

**Comment by nightlark** [comment]
GitHub has a feature for commits with multiple authors: https://help.github.com/articles/creating-a-commit-with-multiple-authors/

**Comment by phlptp** [comment]
@nightlark @kdheepak @trevorhardy   If possible it would be really nice to have this merged and a 2.0 tagged by the time Bryan speaks tomorrow!  @bpalmintier

File: docs/user-guide/filters.md (hunk 1 above)
Comment (phlptp) [reply docs/user-guide/filters.md]: @trevorhardy  did you want me to try to update this section at the risk of it being a little divergent in style?

Comment (trevorhardy) [reply docs/user-guide/filters.md]: I think that would be fine. I would prefer that the content be technically accurate over stylistically consistent. Once you're done I can go back through the stuff you changed and see if I can bring the style back in line.

**Comment by phlptp** [comment]
@trevorhardy  I did some updates for the filters as best as I could.  take a look.

**Comment by phlptp** [comment]
@kdheepak  Once you approve and Trevor gives the OK I think we can merge.

**Pull request merged**